### PR TITLE
upgrade fiesta

### PR DESCRIPTION
**Pull request by twrichards**
https://trello.com/c/jn7vNTvc/308-path-manager-upgrade-fiesta

## What does this change?
https://github.com/guardian/path-manager/pull/29 was failing to deploy in CODE with silent errors (had to SSH onto the box to get the error), [a Play upgrade was suggested](https://github.com/guardian/path-manager/pull/29#issuecomment-663642416) so this is it, along with bumping everything to latest(ish)...

- **Play** `2.4.6` -> **`2.8.2`**
- **Scala** `2.11.6` -> **`2.13.3`**
- **SBT** `0.13.9` -> **`1.3.13`**
- PLUS dependencies to their latest(ish)

## How to test
Run this branch in `CODE` (ideally - so you can test Metrics, Kinesis Logging etc.) and use Composer to create new headlines and change existing headlines - [you should see entries in the logs](https://logs.gutools.co.uk/app/kibana#/discover?_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:now-30m,to:now))&_a=(columns:!(message,level),filters:!(('$state':(store:appState),meta:(alias:!n,disabled:!f,index:a35a6090-59d7-11e8-bbe4-cbb5b151b19c,key:app,negate:!f,params:(query:path-manager),type:phrase,value:path-manager),query:(match:(app:(query:path-manager,type:phrase)))),('$state':(store:appState),meta:(alias:!n,disabled:!f,index:a35a6090-59d7-11e8-bbe4-cbb5b151b19c,key:stage,negate:!f,params:(query:CODE),type:phrase,value:CODE),query:(match:(stage:(query:CODE,type:phrase))))),index:a35a6090-59d7-11e8-bbe4-cbb5b151b19c,interval:auto,query:(language:kuery,query:'%22updated%20canonical%20path%22'),sort:!(!('@timestamp',desc)))) and [Metrics in CloudWatch](https://eu-west-1.console.aws.amazon.com/cloudwatch/home?region=eu-west-1#metricsV2:graph=~(metrics~(~(~'AppMetrics~'pathRegistrations~'App~'path-manager~'Stage~'CODE)~(~'...~'PROD~(yAxis~'right))~(~'.~'pathUpdates~'.~'.~'.~'.~(color~'*23ffbb78~yAxis~'right))~(~'...~'CODE~(color~'*23aec7e8~yAxis~'left)))~view~'timeSeries~stacked~false~region~'eu-west-1~start~'-PT1H~end~'P0D~stat~'Sum~period~60);query=~'*7bAppMetrics*2cApp*2cStage*7d).

## How can we measure success?
This brings things up to latest(ish) which is always good and also unblocks https://github.com/guardian/path-manager/pull/29 🎉 

## Have we considered potential risks?
There are some pretty big leaps in dependencies so there is some inherent risk in that (subtle behaviour changes), that said this is a fairly simple play app and everything seems to work in `CODE`. Would probably appreciate an explicit 👍  from @sihil before merging.

## Images
fiesta

### CODE REVIEW

**Comment by AWare** [comment]
Is this runtime or compiletime DI?

**Comment by twrichards** [comment]
> Is this runtime or compiletime DI?

@AWare it's compile-time (now at least - I eliminated all the `@Inject`s that I introduced in the WIP commits)

I've tested it in `CODE` (including Logging and Metrics which had to change a bit).

File: path-manager/app/AppLoader.scala [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import play.api._
+import play.api.ApplicationLoader.Context
+import play.api.routing.Router
+import play.filters.HttpFiltersComponents
+import router.Routes
+import services.{KinesisLogging, Metrics}
+
+class AppLoader extends ApplicationLoader {
+  def load(context: Context) = {
+
```
Comment (sihil) [inline path-manager/app/AppLoader.scala:7]: Pretty sure you need to initialise logging around here somewhere...
https://www.playframework.com/documentation/2.8.x/ScalaCompileTimeDependencyInjection#Configuring-Logging

File: path-manager/app/AppLoader.scala [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    new KinesisLogging(context.initialConfiguration)
+
+    new AppComponents(context).application
+
+  }
+}
+
+class AppComponents(context: Context) extends BuiltInComponentsFromContext(context) with HttpFiltersComponents {
```
Comment (sihil) [inline path-manager/app/AppLoader.scala]: By convention we split these elsewhere and suggest you do the same here.

**Review by sihil — COMMENTED** [review]
Great work, LGTM other than I suspect you're not going to get any logging at the moment (this catches me out everytime I convert to compile time DI).

File: path-manager/app/AppLoader.scala (hunk 1 above)
Comment (twrichards) [reply path-manager/app/AppLoader.scala:7]: yeah its on the line below this one already ⬇️ 
https://github.com/guardian/path-manager/blob/fece6cb2bb0c2a9e2e44420d7977d79fcc441db2/path-manager/app/AppLoader.scala#L12

Comment (twrichards) [reply path-manager/app/AppLoader.scala:7]: and it works in CODE

**Review by sihil — APPROVED** [review]

**Pull request merged**